training/views.py

```diff
@@ -1103,9 +1103,7 @@ def post(self, request, *args, **kwargs):
     print(eventid)
     event = TrainingEvents.objects.get(id=eventid)
     participantname = self.request.user.first_name+" "+self.request.user.last_name
-
-    teststatus = EventTestStatus.objects.filter(event_id=eventid, fossid=kwargs['testfossid'], mdlemail=self.request.user.email).first()
-
+    teststatus = EventTestStatus.objects.filter(event_id=eventid, fossid=kwargs['testfossid'], mdlemail=self.request.user.email,mdlgrade__gte=40.00).order_by('-mdlgrade').first()
```

Comment:
Is sort by mdlgrade the most appropriate thing? or is timestamp better for latest test score? please check with the training team, if required.

the 40.00 should go into config. it can't be hardcoded here.

put a space after comma for readability.

Reply:
Ideally, if the score is greater than 40.00, then the student cannot retake the test. So this query should give only 1 result.
I have added a commit with other suggested changes : [https://github.com/Spoken-tutorial/spoken-website/pull/461/commits/42fec3908c4c51c725718401845619087c623ffa](url)